src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/GetDocumentByResourceName.cs

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using EdFi.DataManagementService.Core.External.Backend;
+using Microsoft.Extensions.Logging;
+using Npgsql;
+
+namespace EdFi.DataManagementService.Backend.Postgresql.Operation;
+
+public interface IGetDocumentByResourceName
+{
+    public Task<GetResult> GetByResourceName(IGetRequest getRequest, int offset, int limit);
+}
+
+public class GetDocumentByResourceName(NpgsqlDataSource _dataSource, ILogger<GetDocumentByResourceName> _logger) : IGetDocumentByResourceName
+{
+    public async Task<GetResult> GetByResourceName(IGetRequest getRequest, int offset, int limit)
```

Comment:
Your operation should take an `IQueryRequest` and return a `QueryResult`. Those objects handle pagination parameters and multiple document results respectively.